### DISCUSSION TITLE
Regarding the issue with the PVC in the IDE pod.

### DISCUSSION THREAD

**Discussion by wyhdil** [eclipse-theia/theia-cloud#255] [Q&A] [answered]
Hello, I have some questions about PVC that I would like to ask for guidance.

From the previous questions, I learned that this file is a[ YAML template ](https://github.com/eclipsesource/theia-cloud/blob/main/java/operator/org.eclipse.theia.cloud.operator/src/main/resources/templateDeployment.yaml#L43-L96)for creating the IDE pod.
In theory, YAML template should be placeholders for mounting PVC be like: 
![image](https://github.com/eclipsesource/theia-cloud/assets/69418800/6b64be34-3d4d-4761-a874-d753f74f9a9b)
But I haven't found a structure like this in that YAML template.

However, the IDE pod clearly has volume mounts, including PVCs.
![image](https://github.com/eclipsesource/theia-cloud/assets/69418800/b3c59c84-0834-4b93-911b-5692cd166331)

So, I'm curious about how this is achieved, how PVC mounting is implemented.

**Comment by sgraband** [comment, marked as answer]
Hey @wyhdil,

While we're here to assist, we encourage users to delve into the code for straightforward issues. In this instance, a quick code search for `user-data` would lead you directly to [this code](https://github.com/eclipsesource/theia-cloud/blob/e825f4aef4778791869212bfb29d1d551873ff52/java/operator/org.eclipse.theia.cloud.operator/src/main/java/org/eclipse/theia/cloud/operator/handler/impl/LazySessionHandler.java#L451), which should answer your query. 

Remember, this is a free support channel and detailed code walkthroughs aren't always possible. Please do reach out for any unclear aspects or poorly documented code.

**Reply by wyhdil** [reply]
Okay, thank you for providing the solution to the problem.
